framework/src/mesh/MooseMesh.C

```diff
@@ -738,10 +742,10 @@ MooseMesh::cacheInfo()
   }
 }
 
-std::set<SubdomainID> &
-MooseMesh::getNodeBlockIds(const Node & node)
+const std::set<SubdomainID> &
+MooseMesh::getNodeBlockIds(const Node & node) const
 {
-  return _block_node_list[node.id()];
+  return _block_node_list.at(node.id());
```

Comment:
I know you used `map::at` here because it has a const version, but this subtly changes the runtime behavior of this function when `node.id()` is not found.  Before, it would return a default-constructed set, now it will throw a `std::out_of_range` exception...

MO we should try to `find()` the relevant key and then either `mooseError` or insert a newly-constructed set, depending on which behavior is correct in the larger scheme of things.